python/tank/bootstrap/manager.py

```diff
@@ -236,17 +298,62 @@ def bootstrap_engine(self, engine_name, entity=None):
         the engine may not be the same as the API version that was
         executed during the bootstrap.
 
-        :param entity: Shotgun entity to launch engine for
-        :type entity: Dictionary with keys type and id
+        :param engine_name: Name of engine to launch (e.g. ``tk-nuke``).
+        :param entity: Shotgun entity to launch engine for.
+        :param progress_callback: Callback function that reports back on the toolkit and engine bootstrap progress.
+        :param completed_callback: Callback function that handles cleanup after successful completion of the bootstrap.
+        :param failed_callback: Callback function that handles cleanup after failed completion of the bootstrap.
+        :returns: :class:`sgtk.platform.Engine` instance when ``async_bootstrap` is false; otherwise None.
+        """
+
+        log.info("Bootstrapping engine %s for entity %s." % (engine_name, entity))
+
+        bootstrap_asynchronously = self.async_bootstrap
+
+        if bootstrap_asynchronously:
+            try:
+                import threading
+            except ImportError:
+                log.warning("Cannot bootstrap asynchronously in a background thread.")
+                # Since Qt is not available, fall back on synchronous boostraping.
+                bootstrap_asynchronously = False
+
+        if bootstrap_asynchronously:
```

Comment:
The structure of this method makes me think 2 methods might be better unless there's a reason to keep them as one.